document/json/encoder.go

```diff
@@ -324,3 +324,10 @@ func isValidJSONNumber(s string) bool {
 	// Make sure we are at the end.
 	return s == ""
 }
+
+func (e *Encoder) unsupportedType(v interface{}) error {
+	if document.IsNoSerde(v) {
+		return &document.InvalidUnmarshalError{Type: reflect.TypeOf(v)}
```

Comment:
is this backwards incompatible? should i be performing the same error behavior as L24?